Subject: [PATCH] fix numpy distutils
---
Index: waftools/python_cfg.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/waftools/python_cfg.py b/waftools/python_cfg.py
--- a/waftools/python_cfg.py	(revision f31604581eaaa2b4524b7f9d89e988b85f896e27)
+++ b/waftools/python_cfg.py	(date 1735891614876)
@@ -71,8 +71,8 @@
     self.start_msg("Checking for numpy include")
     # retrieve includes dir from numpy module
     numpy_includes = self.get_python_variables(
-        ['"\\n".join(misc_util.get_numpy_include_dirs())'],
-        ["from numpy.distutils import misc_util"],
+        ['"\\n".join([np.get_include()])'],
+        ["import numpy as np"],
     )
     if self.is_defined("ASTER_PLATFORM_MINGW"):
         incs = [PureWindowsPath(i) for i in numpy_includes]
